### DISCUSSION TITLE
Python API not found after fresh installation

### DISCUSSION THREAD

**Discussion by LukasPoque** [osm-search/Nominatim#3471] [General]
So I started with a fresh installation of Ubuntu 22.04.4 LTS on a bare metal server and then followed the steps from here: https://nominatim.org/release-docs/latest/appendix/Install-on-Ubuntu-22/

Since I followed these instructions without importing any data I thought it was normal that `http://localhost/nominatim/` returns a 404 Error. After setting this up I didn't see any strange errors in the services, so I started importing a dataset like described here: https://nominatim.org/release-docs/latest/admin/Import/. I didn't realize that the project folder there was different from the one used in the https://nominatim.org/release-docs/latest/appendix/Install-on-Ubuntu-22/ tutorial, so I went with `nominatim-planet`. The import worked again without any errors and I can execute a query like `nominatim search --query Berlin` without an error and with a normal answer.

Now to my problem: After importing `http://localhost/nominatim/` is still showing a 404 error, so I realized that in the installation tutorial there was a different path used in the `/etc/systemd/system/nominatim.service` (`WorkingDirectory=$USERHOME/nominatim-project`) then in the import tutorial I followed. I tried to change the path and restarted the nominatim.service but the error is still there.

Any hints where I can get a little bit more information on where the error is? Checking the nginx config I saw there should be a folder `$USERHOME/nominatim-project/website` but there is no folder called `website` in my project folder.

Happy for any hints, this is my first time using Nominatim :)

**Comment by mtmail** [comment]
Nominatim is migrating from PHP to Python, for some versions both frontends can be installed and we keep finding small documentation issues. The nominatim-project vs nominatim-planet was resolved in https://github.com/osm-search/Nominatim/pull/3403

Don't move/rename the project directory after the import. I've done that once and found some configuration inside the database (`osm2pgsql_properties` table) contain the full path of the project directory.

The `root` path in the nginx configuration shouldn't have any impact I think. We'll remove the '/website' part though because that was only needed for the PHP frontend (it used to contain *.php files). https://github.com/osm-search/Nominatim/pull/3472

Nginx forwards all requests to the process that listens on the `proxy_pass` address. I'd expect the HTTP 404 to come from that service. Does the gunicorn logfile contain any errors?

Which path are you using in you nginx configuration? `location /` (as mentioned on https://nominatim.org/release-docs/latest/admin/Deployment-Python/ or `location /nominatim/` (as mentioned on https://nominatim.org/release-docs/latest/appendix/Install-on-Ubuntu-22/) ? It might be as simple as `http://localhost/nominatim/status` vs `http://localhost/status`

**Reply by lonvia** [reply]
It might even be as simple as `http://localhost/nominatim/` vs. `http://localhost/nominatim/status`. There is no default page for the Python webserver.

**Reply by LukasPoque** [reply]
Yes 🎉 I should have tested a lot more with different URLs, `http://localhost/nominatim/status` is working, and `/nominatim/search?` too

Thanks to both of you for your time
